vignettes/_translation_links.R

```diff
@@ -0,0 +1,26 @@
+# build a link list of alternative languages (may be character(0))
+# idea is to look like 'Other languages: en | fr | de'
+.write.translation.links <- function(fmt) {
+    url = "https://rdatatable.gitlab.io/data.table/articles"
+    path = dirname(knitr::current_input(TRUE))
+    if (basename(path) == "vignettes") {
+      lang = "en"
+    } else {
+      lang = basename(path)
+      path = dirname(path)
+    }
+    translation = dir(path,
+      recursive = TRUE,
+      pattern = glob2rx(knitr::current_input(FALSE))
+    )
+    transl_lang = ifelse(dirname(translation) == ".", "en", dirname(translation))
+    block = if (!all(transl_lang == lang)) {
+      sprintf(
+        fmt, paste(collapse = " | ", sprintf(
+          "[%s](%s)",
+          transl_lang[transl_lang != lang],
+          file.path(url, sub("(?i)\\.Rmd$", ".html", translation[transl_lang != lang]))
+      )))
```

Comment:
(untested) I think this will get the effect of including the current language unlinked

```suggestion
      linked_transl = sprintf("[%s](%s)", transl_lang, file.path(url, sub("(?i)\\.Rmd$", ".html", translation))
      linked_transl[transl_lang == lang] = lang
      sprintf(fmt, paste(linked_transl, collapse = " | "))
```

Reply:
I tested your suggestion succesfully, this is what looks like (pkgdown build)
![image](https://github.com/user-attachments/assets/26b88b99-405c-4334-aaab-6cfc24a6197b)
![image](https://github.com/user-attachments/assets/788fa81a-6304-441f-a9d2-da65c05f9529)
![image](https://github.com/user-attachments/assets/418c2415-b898-499b-b93b-fec042516810)

Link's urls are absolute, as they are expected to be followed from vignettes in a path where translations are not available. I don't know whether it is possible to specify to generate relative paths when built on pkgdown but keep absolute paths otherwise. 

Also: if there is no translation available, no message is generate at all (which I think it's ok).

Note. Missing ) in line 18.
```suggestion
      linked_transl = sprintf("[%s](%s)", transl_lang, file.path(url, sub("(?i)\\.Rmd$", ".html", translation)))
      linked_transl[transl_lang == lang] = lang
      sprintf(fmt, paste(linked_transl, collapse = " | "))
```